File: frontend/cypress/test-solutions/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0;
        unordered_map<int, int> sumCount;
        sumCount[0] = 1; // prefix sum -> count
        int prefixSum = 0;
        
        for (int num : nums) {
            prefixSum += num;
            
            // If (prefixSum - k) exists, it means there's a subarray ending here with sum k
            if (sumCount.find(prefixSum - k) != sumCount.end()) {
                count += sumCount[prefixSum - k];
            }
            
            // Update prefix sum count
            sumCount[prefixSum]++;
        }
        
        return count;
    }
};

File: frontend/cypress/test-solutions/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: list[int], k: int) -> int:
        count = 0
        sum_count = {0: 1}  # prefix sum -> count
        prefix_sum = 0
        
        for num in nums:
            prefix_sum += num
            
            # If (prefix_sum - k) exists, it means there's a subarray ending here with sum k
            if prefix_sum - k in sum_count:
                count += sum_count[prefix_sum - k]
            
            # Update prefix sum count
            sum_count[prefix_sum] = sum_count.get(prefix_sum, 0) + 1
        
        return count